rasa/graph_components/adders/slot_extraction_to_history_adder.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+import logging
+
+import rasa.core.actions.action
+from typing import Dict, Text, Any, Optional
+
+from rasa.core.channels.channel import (
+    OutputChannel,
+    CollectingOutputChannel,
+)
+from rasa.core.nlg import NaturalLanguageGenerator
+from rasa.engine.graph import GraphComponent, ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.utils.endpoints import EndpointConfig
+
+from rasa.shared.core.trackers import DialogueStateTracker
+from rasa.shared.core.domain import Domain
+from rasa.shared.core.constants import ACTION_EXTRACT_SLOTS
+
+logger = logging.getLogger(__name__)
+
+
+class SlotExtractionToHistoryAdder(GraphComponent):
+    """Adds slot extractions to DialogStateTracker."""
+
+    @classmethod
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> SlotExtractionToHistoryAdder:
+        """Creates component (see parent class for full docstring)."""
+        return cls()
+
+    async def add(
+        self,
+        tracker: DialogueStateTracker,
+        domain: Domain,
+        nlg: NaturalLanguageGenerator,
+        endpoint_config: Optional[EndpointConfig] = None,
```

Comment:
It's best to not have this as an optional argument, it's required to have the EndpointConfig to run any custom actions if present in the domain 🤔